core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java

```diff
@@ -37,6 +42,37 @@
         this.relationName = relationName;
     }
 
+    @Override
+    public ShapeRelation readFrom(StreamInput in) throws IOException {
```

Comment:
but it feels like we are back with our own ordinal then, more LOC too with two switches compared to the previous `Operator` impl :) can't we just handle bw comp once we actually change the enum? of course the main issue is remembering about this, and tests are a good way to remember...